.github/workflows/release.yml

```diff
@@ -24,6 +24,10 @@ jobs:
         uses: actions/checkout@v4
         with:
           fetch-depth: 0 # Fetches all history for all branches and tags
+          # note: this token should have admin permissions or else the last step
+          # to rebase next will not succeed because of the rulesets protecting
+          # the `next` branch.
+          token: ${{ secrets.GH_TOKEN }}
```

Comment:
We should not be using Personal Access Tokens as they are incredibly insecure and their consequence if leaked are wide (you can reuse the token and escalate privileges if leaked), we should be using [an GitHub App](https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/differences-between-github-apps-and-oauth-apps#machine-vs-bot-accounts), similar to [Ecospark](https://github.com/apps/ecospark), [Ecoscript](https://github.com/apps/ecoscript), as explained in [github-app-token](https://github.com/tibdex/github-app-token#github-app-token).

[More detailed docs.](https://web.archive.org/web/20230115194214/https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/automating-projects-using-actions#example-workflow-authenticating-with-a-github-app)

Reply:
heard. i'll close and reopen with a new implementation